lib/events.js

```diff
@@ -437,9 +441,9 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners) {
         // LIFO order
-        do {
-          this.removeListener(type, listeners[listeners.length - 1]);
-        } while (listeners[0]);
+        for (i = listeners.length; i-- > 0;) {
```

Comment:
Oh wow. Would love to see some feedback from one or two other @nodejs/collaborators on this. You're re-introducing this pattern here. Do we really want to embrace that pattern or move away from it?